File: Clojure/src/clojure/pi_quadrature/pi_sequential_loop.clj
;  Calculation of π using quadrature. Sequential algorithm.
;
;  Copyright © 2009–2011, 2013–2015  Russel Winder

(ns pi-quadrature.pi-sequential-loop)

(load "output")

(defn summation [count delta]
  (loop [
          i count
          s 0.0
         ]
    (if (zero? i)
      s
      (let [x (* (- i 0.5) delta)]
        (recur (dec i) (+ s (/ 1.0 (+ 1.0 (* x x)))))))))

(defn -main []
  (let [
        n  100000000 ; 10 times fewer than Java due to speed issues.
        delta  (/ 1.0 n)
        startTimeNanos  (System/nanoTime)
        pi  (* 4.0 delta (summation n delta))
        elapseTime  (/ (- (System/nanoTime) startTimeNanos) 1e9)
        ]
    (out "Sequential Loop" pi n elapseTime)))

File: Kotlin/src/main/kotlin/uk/org/winder/pi_quadrature/pi_sequential_loop.kt
/*
 *  Calculation of π using quadrature realized with sequential for loop.
 *
 *  Copyright © 2013–2015, 2018  Russel Winder
 */

package uk.org.winder.pi_quadrature

fun main(args:Array<String>) {
	val n = 1_000_000_000
	val delta = 1.0 / n
	val startTimeNanos = System.nanoTime()
	var sum = 0.0
	for (i in 1..n) {
		val x = (i - 0.5) * delta
		sum += 1.0 / (1.0 + x * x)
	}
	val pi = 4.0 * delta * sum
	val elapseTime = (System.nanoTime() - startTimeNanos) / 1e9
	out("Sequential Loop", pi, n, elapseTime)
}
